cli/CommandExecutor.cpp

```diff
@@ -0,0 +1,185 @@
+/**
+ *   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+ *     University of Wisconsin—Madison.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include "cli/CommandExecutor.hpp"
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdio>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "catalog/CatalogAttribute.hpp"
+#include "catalog/CatalogDatabase.hpp"
+#include "catalog/CatalogRelation.hpp"
+#include "catalog/CatalogRelationSchema.hpp"
+#include "cli/PrintToScreen.hpp"
+#include "parser/ParseStatement.hpp"
+#include "utility/PtrVector.hpp"
+#include "utility/Macros.hpp"
+#include "utility/SqlError.hpp"
+
+#include "gflags/gflags.h"
+#include "glog/logging.h"
+
+using std::fprintf;
+using std::fputc;
+using std::fputs;
+using std::size_t;
+using std::string;
+using std::vector;
+
+namespace quickstep {
+namespace cli {
+namespace {
+
+namespace C = ::quickstep::cli;
+
+void executeDescribeDatabase(
+    const PtrVector<ParseString> *arguments,
+    const CatalogDatabase &catalog_database, FILE *out) {
+  // Column width initialized to 6 to take into account the header name
+  // and the column value table
+  int max_column_width = C::kInitMaxColumnWidth;
+  const CatalogRelation *relation;
+  if (arguments->size() == 0) {
+    for (const CatalogRelation &rel : catalog_database) {
+      int column_width = static_cast<int>(rel.getName().length());
+      max_column_width =
+          max_column_width < column_width ? column_width : max_column_width;
+    }
+  } else {
+    const ParseString &table_name = arguments->front();
+    const std::string &table_name_val = table_name.value();
+    relation = catalog_database.getRelationByName(table_name_val);
+
+    if (relation == nullptr) {
+     THROW_SQL_ERROR_AT(&(arguments->front())) << " Unrecognized relation "  <<table_name_val;
```

Comment:
Please check the indentation before `THROW_SQL_ERROR_AT` and whitespace around `<< table_name_val`.